rebasehelper/tests/test_application.py

```diff
@@ -112,16 +128,48 @@ def test_application_sources(self, workdir):
                                      False]}},
             'results_dir': os.path.join(workdir, constants.RESULTS_DIR)}
 
-        cli = CLI(self.cmd_line_args)
-        config = Config()
-        config.merge(cli)
-        execution_dir, results_dir, debug_log_file = Application.setup(config)
-        app = Application(config, execution_dir, results_dir, debug_log_file)
         app.prepare_sources()
         for key, val in app.kwargs.items():
             if key in expected_dict:
                 assert val == expected_dict[key]
 
+    def test_setup(self, config):
+        config = config(self.cmd_line_args)
+        files = Application.setup(config)
+        for file in files:
+            assert os.path.exists(file)
+
+    def test_setup_continue(self, config):
+        config = config(self.cmd_line_args + ['--continue'])
+        files = Application.setup(config)
+        for file in files:
+            assert os.path.exists(file)
+
+        # check that if rebase-helper-results exists, it does not get removed
+        test_file = os.path.join(files[1], 'test')
+        open(test_file, 'w').close()
+        Application.setup(config)
+        assert os.path.exists(test_file)
+
+    def test_rpm_packages(self, app):
+        from rebasehelper.results_store import results_store
```

Comment:
Why not global import?

Reply:
I am not completely sure how pytest namespaces work. What would happen if there were 2 tests that affect `results_store`? Wouldn't the changes from the first test be present even in the second test? I expect the tests to be independent of each other.

Reply:
And how exactly does this prevent that?

First of all, `results_store` is instantiated already when application.py imports it, and second, it's cleared in `Application` constructor.